### ISSUE TITLE
Seat extension

### ISSUE THREAD

**Issue by skamoen** [AreaFiftyLAN/lancie-api#11]
Seats need more functionality. Should have a Label and several configuration options. (add rows, ect)

Team captains should be able to reserve seats for their entire team

**Comment by bvanderdrift** [comment]
We willen wel dat individueel nog mogelijk is.

Daarnaast, willen we dat teamcaptains losse plekken selecteerd voor elk teamlid? Of zoals bij pathe kom je altijd 4 op een rij?

Ook: Als een teamcaptain teamplekken kiest, maar één van de teamleden heeft al een plek gekozen. Mag de teamcaptain hem dan aanpassen?

Wat als iemand in meerdere teams zit ivm meerdere toernooien?

**Comment by skamoen** [comment]
Seems to me like all captains of teams of which the user is a part are allowed to reserve a seat, as well as the user itself. I highly doubt a captain will abuse this. If so, he won't be a captain anymore. 

It makes sense to reserve spots in a row, but I dont think this should be a hard constraint. That is something which can be done in the front-end though.

**Comment by skamoen** [comment]
Alright. Time to figure this out properly.

What seats need to do, looking at it from the front-end:

- [ ] Find where a user is seated
- [ ] See which seats are available
- [ ] See which User reserved a specific seat
- [ ] Allow Team Captains to also reserve seats for their Team. Initially there are no constraints. Captains can put their teammembers anywhere they want. I doubt this will be abused. If so, we could make a Captain Blacklist, integrated into the Authorize service. 

So basically, Seats need to be retrievable from the database by Username and by Location. A location consists of a "groupname" and a seatnumber. 

Authorization needs to be either an Admin, User, or Captain of a Team of which the User is a member.

From a backend perspective, you need to be able to add seats to the seatmap. This can be done by specifying the groupname and the number of seats in this group. 

Anything else?
